packages/styled-system/src/utils/parse-gradient.ts

```diff
@@ -64,7 +64,12 @@ export function parseGradient(value: string | null | undefined, theme: Dict) {
     // else, get and transform the color token or css value
     const key = `colors.${_color}`
     const color = key in theme.__cssMap ? theme.__cssMap[key].varRef : _color
-    return _stopOrFunc ? [color, _stopOrFunc].join(" ") : color
+    return _stopOrFunc
```

Comment:
Can we add a brief comment that explains what this operation does?

Reply:
This change will revert and break what I did in #4460 - Correct parseGradient function so that it correctly handles CSS functions.

Reply:
As I stated in the summary, that operation simply flattens the array.
The fix will not break #4460, as my implementation passes all tests, including the ones you added. Instead #4460 introduced the problem.

`radial(farthest-side at 50% 50%, black, white)` produces `radial-gradient(farthest-side at,50%,50%, black, white)` <-- additional commas.
This happens because `_stopOrFunc` is not a string nor a func, but it's an array. The evaluations:
```js
color => 'farthest-corner';
_stopOrFunc => [ 'at', '50%', '50%' ];

[color, _stopOrFunc].join(" ") => ['farthest-corner', [ 'at', '50%', '50%' ]].join(" ") => "farthest-corner at,50%,50%]"
```
that because JS auto flattens the inner array with the default comma separator (you can check the docs).

So what I did, is simply check if the `_stopOrFunc` is an array. And if it is, I flatten it to
`['farthest-corner, 'at', '50%', '50%']` <-- NO INNER array
so the join operation actually works.

Reply:
Have you tried to access the CSB that was created for this? It doesn't work for me. There are usually two rows, but here there's only one. This is the typical display:

![image](https://user-images.githubusercontent.com/2768849/141166892-1acf631d-b552-45e7-98ce-a6c03d60de43.png)

 The CSB is having a problem loading `https://unpkg.com/@chakra-ui/provider@1.6.10`. Version 1.6.9 is there, but not 1.6.10. Also there's a `@chakra-ui/machine` in the package.json. Any idea how that got there?

Reply:
Sorry, don't know what you referring to.
I just fixed the function. I didn't touch anything else.